python/ray/serve/tests/test_custom_metrics.py

```diff
@@ -0,0 +1,117 @@
+import sys
+from typing import Any, Dict
+
+import pytest
+
+import ray
+from ray import serve
+from ray._common.test_utils import SignalActor, wait_for_condition
+from ray.serve._private.common import DeploymentID
+from ray.serve._private.test_utils import (
+    send_signal_on_cancellation,
+)
+
+
+def get_autoscaling_metrics_from_controller(
+    client, deployment_id: DeploymentID
+) -> Dict[str, float]:
+    """Get autoscaling metrics from the controller for testing."""
+    ref = client._controller._dump_all_autoscaling_metrics_for_testing.remote()
+    metrics = ray.get(ref)
+    return metrics.get(deployment_id, {})
+
+
+class TestCustomServeMetrics:
+    """Check that redeploying a deployment doesn't reset its start time."""
+
+    def test_custom_serve_metrics(self, serve_instance):
+        @serve.deployment(
+            autoscaling_config={
+                "min_replicas": 1,
+                "max_replicas": 5,
+                "target_num_ongoing_requests_per_replica": 2,
+                "upscale_delay_s": 2,
+                "downscale_delay_s": 10,
+            }
```

Comment:
currently on replica push frequency is controlled by `metrics_interval_s` not RAY_SERVE_REPLICA_AUTOSCALING_METRIC_PUSH_INTERVAL_S. By setting `metrics_interval_s=1` or small value your test will run faster, without this it will use the default `10s` delay

```suggestion
                "downscale_delay_s": 10,
                "metrics_interval_s": 1
            }
```

Reply:
same comment in other tests